### PR TITLE
Fairscale integration tests for Lite

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Analogous to #14901 and #14815 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @carmocca @justusschock @awaelchli @borda @akihironitta @rohitgr7

### CODE REVIEW

File: src/lightning_lite/strategies/fairscale.py
Comment (awaelchli) [inline src/lightning_lite/strategies/fairscale.py:62]: These silly mistakes are now covered by the tests.

File: tests/tests_pytorch/strategies/test_sharded_strategy.py
Comment (awaelchli) [inline tests/tests_pytorch/strategies/test_sharded_strategy.py:21]: TODO: remove before merge

File: src/lightning_lite/strategies/fairscale.py
Comment (awaelchli) [inline src/lightning_lite/strategies/fairscale.py:183]: this was duplicated dead code from earlier refactor, the same function is already there below

File: src/lightning_lite/lite.py
Comment (carmocca) [inline src/lightning_lite/lite.py:164]: debugging comment

File: src/lightning_lite/strategies/fairscale.py
Comment (carmocca) [inline src/lightning_lite/strategies/fairscale.py:183]: Can you also delete the definitions in `sharded` and `sharded_spawn` to use the function below?

**Review by carmocca — APPROVED** [review]

File: src/lightning_lite/lite.py
Comment (awaelchli) [reply src/lightning_lite/lite.py]: yeah, i wasn't finished debugging, but should be fixed now

File: src/lightning_lite/strategies/fairscale.py
Comment (awaelchli) [reply src/lightning_lite/strategies/fairscale.py:183]: I'm hesitant to do so. It would mean touching src/pytorch_lightning, increasing the codeowner list.

**Review by justusschock — APPROVED** [review]

File: src/lightning_lite/strategies/fairscale.py
Comment (carmocca) [reply src/lightning_lite/strategies/fairscale.py:183]: You are still blocked by more CODEOWNERs after two approvals, so might as well go for it :)

Comment (justusschock) [reply src/lightning_lite/strategies/fairscale.py:183]: @awaelchli It shouldn't too much. For strategy-related changes, you should be fine with @carmocca and myself as reviewers (as I am codeowner for strategies and we only need one codeowner approval and two in total).

Comment (carmocca) [reply src/lightning_lite/strategies/fairscale.py:183]: NVM my comment above. They are only marked as codeowner because of the comments in the PL tests that will be reverted. Feel free to resolve and I'll do this in a follow-up

**Review by otaj — APPROVED** [review]

File: src/lightning_lite/strategies/fairscale.py
Comment (carmocca) [reply src/lightning_lite/strategies/fairscale.py:183]: Codeowners are no longer a blocker

**Pull request merged**
